### ISSUE TITLE
BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by yutayamazaki** [yutayamazaki/paper_reading#24]
### 論文
[BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding](https://arxiv.org/abs/1810.04805)

### 著者・所属機関
- Jacob Devlin, Ming-Wei Chang, Kenton Lee, Kristina Toutanova
- Google AI Language

### 投稿日
- 2018/10/11

## Abstract
- Bidirectional Encoder Representations from Transformers; BERT
- 事前学習モデルとして汎用的に利用可能で，CVにおけるResNetのようなもの
- この事前学習モデルを利用して11のタスクでSOTA(しかもまあまあ大幅に上回った)
- 事前学習のアプローチは大きく分けてfeature-basedとfine-tuningの2つ
- feature-basedはELMoに代表される手法で，各タスクに特化したアーキテクチャを持ち，fine-tuningはOpenAI GPTに代表される手法で，各タスクに特化したパラメータの数が少ない(学習時間が早い)手法
- BERTはfine-tuningアプローチのひとつ

## 先行研究と比べて何がすごい？
- 基本モデルはTransformer
- 少しの学習手法の変更だけでここまで汎用的にfine-tuningに用いれることがすごい

### 学習方法
- 入力データ
    - トークン，セグメント，positionの3つをエンコーディングしたものの和を入力とする
- Masked LMとNext Sentence Predictionという2つのタスクのlossの和で学習する

### Masked LM
- 単語列を15%の確率でランダムにマスクし，そのマスクした単語が何かを予測するタスクを解いて学習させる
- マスクの手順は
    - 80%の確率でそのままマスクする
    - 10%の確率で選択した単語を別の単語に置き換える
    - 10%の確率で置き換えない

### Next Sentence Prediction
- Question AnsweringやNatural Language Inferenceでは2つの文章間の関係性が重要になるが，それはMasked LMでは学習できない
- それを学習するため，ある文章のペアが与えられた時に，文章Aの次に続くのが文章Bであるかどうかを2値分類する
- Bは50%の確率で次の文章で，残り50%の確率でランダムにサンプリングした文章

## どうやって有効だと検証した？
- めちゃめちゃSOTA
